SwiftPrivilegedHelperApplication/Scripts/CodeSignUpdate.swift

```diff
@@ -0,0 +1,400 @@
+func updateSMAuthorizedClients(
+    in plistDict: inout [String: AnyObject],
+    with s: String)
+{
+    assert(target == "helper")
+    guard let prodBundleID = plistDict["CFBundleIdentifier"] as? String else
+    {
+        emitError(
+            "Helper info property list, \(infoPListFile), is missing "
+            + "\"CFBundleIdentifier\" key, or it not a string"
+        )
+    }
+    guard prodBundleID == bundleIdentifierHelper else
+    {
+        emitError(
+            "Bundle id in info propery list, \(infoPListFile), does not"
+            + " match HELPER_BUNDLE_ID\n"
+            + "     plists CFBundleIdentifier = \(prodBundleID)\n"
+            + "              HELPER_BUNDLE_ID = \(bundleIdentifierHelper)"
+        )
+    }
+    
+    guard let prodBundleName = plistDict["CFBundleName"] as? String else
+    {
+        emitError(
+            "Helper info property list, \(infoPListFile), is missing "
+            + "\"CFBundleName\" key, or it not a string"
+        )
+    }
+    guard prodBundleName == bundleIdentifierHelper else
+    {
+        emitError(
+            "Bundle name in info propery list, \(infoPListFile), does "
+            + "not match HELPER_BUNDLE_ID\n"
+            + "     plists CFBundleName = \(prodBundleName)\n"
+            + "        HELPER_BUNDLE_ID = \(bundleIdentifierHelper)"
+        )
+    }
+    
+    plistDict.removeValue(forKey: "SMAuthorizedClients")
+    let newClients: [String] = [s]
+    plistDict["SMAuthorizedClients"] = newClients as NSArray
+}
+
+guard let action = Environment["ACTION"] else {
+    environmentVarNotSet("ACTION")
+}
+
+var appString = ""
+var helperString = ""
+
+switch action
+{
+    case "build":
+        appendApplicationBundleIdentifier(to: &appString)
+        appString += " and "
+        appendAppleGeneric(to: &appString)
+        appString += " and "
+        appendMacDeveloper(to: &appString)
+        appString += " and "
+        appendAppleMacDeveloper(to: &appString)
+        appString += " /* exists */"
+    
+        appendHelperBundleIdentifier(to: &helperString)
+        helperString += " and "
+        appendAppleGeneric(to: &helperString)
+        helperString += " and "
+        appendMacDeveloper(to: &helperString)
+        helperString += " and "
+        appendAppleMacDeveloper(to: &helperString)
+        helperString += " /* exists */"
+    
+    case "install":
+        appendAppleGeneric(to: &appString)
+        appString += " and "
+        appendApplicationBundleIdentifier(to: &appString)
+        appString += " and "
+        appendAppleDeveloperID(to: &appString)
+        appString += " and "
+        appendDeveloperID(to: &appString)
+    
+        appendAppleGeneric(to: &helperString)
+        appString += " and "
+        appendHelperBundleIdentifier(to: &helperString)
+        appString += " and "
+        appendAppleDeveloperID(to: &helperString)
+        appString += " and "
+        appendDeveloperID(to: &helperString)
```

Comment:
I think that there are typos here...these should probably be `helperString` instead of `appString`:

```suggestion
        appendAppleGeneric(to: &helperString)
        helperString += " and "
        appendHelperBundleIdentifier(to: &helperString)
        helperString += " and "
        appendAppleDeveloperID(to: &helperString)
        helperString += " and "
        appendDeveloperID(to: &helperString)
```